File: tutor-claude/msa-tutor/common/src/main/java/be/com/msatutor/common/api/ApiError.java
package be.com.msatutor.common.api;

import java.time.Instant;
import java.util.List;

public record ApiError(
    Instant timestamp,
    int status,
    String error,
    String message,
    String path,
    List<FieldError> errors
) {
    public record FieldError(String field, Object rejectedValue, String message) {
    }
}

File: tutor-claude/spring-security-tutor/src/main/kotlin/be/com/springsecuritytutor/common/ApiError.kt
package be.com.springsecuritytutor.common

import java.time.Instant

data class ApiError(
    val timestamp: Instant = Instant.now(),
    val status: Int,
    val error: String,
    val message: String,
    val path: String?,
    val errors: List<FieldError> = emptyList()
) {
    data class FieldError(
        val field: String,
        val rejectedValue: Any?,
        val message: String
    )
}
